File: Python/242. Valid Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        hashCount = {}
        for i in range(len(s)):
            if s[i] not in hashCount: hashCount[s[i]] = 1
            else: hashCount[s[i]] += 1
            if t[i] not in hashCount: hashCount[t[i]] = -1
            else: hashCount[t[i]] -= 1
        
        for val in hashCount.values():
            if val != 0:
                return False
        return True
    
print(Solution().isAnagram("anagram", "nagaram"))

File: TypeScript/242. Valid Anagram.ts
function isAnagram(s: string, t: string): boolean {
    const hashS: {[key: string]: number} = {};
    const hashT: {[key: string]: number} = {};
    
    if (s.length !== t.length)
        return false;

    for (let i = 0; i < s.length; i++) {
        if (!(s[i] in hashS)) hashS[s[i]] = 1;
        else hashS[s[i]] += 1;
        
        if (!(t[i] in hashT)) hashT[t[i]] = 1;
        else hashT[t[i]] += 1;
    }

    for (const key in hashS) {
        if (hashS[key] !== hashT[key]) {
            return false;
        }
    }
    
    return true;
};

console.log(isAnagram("anagram", "nagaram"))